baggage/baggage.go

```diff
@@ -33,14 +33,15 @@ const (
 	keyValueDelimiter = "="
 	propertyDelimiter = ";"
 
+	// if you update these 2 values you must update the static implementation below: keyReValidator and valReValidator
 	keyDef      = `([\x21\x23-\x27\x2A\x2B\x2D\x2E\x30-\x39\x41-\x5a\x5e-\x7a\x7c\x7e]+)`
 	valueDef    = `([\x21\x23-\x2b\x2d-\x3a\x3c-\x5B\x5D-\x7e]*)`
 	keyValueDef = `\s*` + keyDef + `\s*` + keyValueDelimiter + `\s*` + valueDef + `\s*`
 )
 
 var (
-	keyRe      = regexp.MustCompile(`^` + keyDef + `$`)
-	valueRe    = regexp.MustCompile(`^` + valueDef + `$`)
+	keyRe      = keyReValidator{}
+	valueRe    = valReValidator{}
 	propertyRe = regexp.MustCompile(`^(?:\s*` + keyDef + `\s*|` + keyValueDef + `)$`)
```

Comment:
How about removing this regex as well?

Reply:
This regex is slightly more complex and it is actually used to get subgroups:

```
match := propertyRe.FindStringSubmatch(property)
	if len(match) != 4
```

The ones I changed are simple and only used for validation.

Reply:
This is true. However, I expect it could be a significant improvement for `Parse`. Is it something that you would like to take a look (as a separate PR)?

Reply:
Working on it.